File: Sources/MCPManagerMac/Services/ToolRegistry.swift
import Foundation

enum ToolRegistry {
    static func tools(forWorkspaceRoot workspaceRoot: String) -> [CLITool] {
        let home = FileManager.default.homeDirectoryForCurrentUser.path
        let workspaceURL = URL(fileURLWithPath: workspaceRoot, isDirectory: true)

        return [
            CLITool(
                id: "codex-cli",
                name: "Codex CLI",
                cliCommand: "codex",
                format: .codexToml,
                scope: .user,
                officialPath: URL(fileURLWithPath: home)
                    .appendingPathComponent(".codex", isDirectory: true)
                    .appendingPathComponent("config.toml", isDirectory: false)
                    .path,
                docsURL: "https://platform.openai.com/docs/guides/tools-remote-mcp",
                verifyArgs: ["mcp", "list"]
            ),
            CLITool(
                id: "claude-code",
                name: "Claude Code",
                cliCommand: "claude",
                format: .claudeProjectJSON,
                scope: .project,
                officialPath: workspaceURL.appendingPathComponent(".mcp.json", isDirectory: false).path,
                docsURL: "https://docs.anthropic.com/en/docs/claude-code/mcp",
                verifyArgs: ["mcp", "list"]
            ),
            CLITool(
                id: "gemini-cli",
                name: "Gemini CLI",
                cliCommand: "gemini",
                format: .geminiProjectJSON,
                scope: .project,
                officialPath: workspaceURL
                    .appendingPathComponent(".gemini", isDirectory: true)
                    .appendingPathComponent("settings.json", isDirectory: false)
                    .path,
                docsURL: "https://google-gemini.github.io/gemini-cli/docs/tools/mcp-server.html",
                verifyArgs: ["mcp", "list", "--scope", "project"]
            )
        ]
    }
}

File: src/main/services/toolRegistry.js
const os = require('os');
const path = require('path');

function buildOfficialTools(workspaceRoot = process.cwd()) {
  const home = os.homedir();

  return [
    {
      id: 'codex-cli',
      name: 'Codex CLI',
      cliCommand: 'codex',
      format: 'codex_toml',
      scope: 'user',
      officialPath: path.join(home, '.codex', 'config.toml'),
      configPath: path.join(home, '.codex', 'config.toml'),
      pathLocked: true,
      docsUrl: 'https://platform.openai.com/docs/docs-mcp',
      verifyArgs: ['mcp', 'list'],
    },
    {
      id: 'claude-code',
      name: 'Claude Code',
      cliCommand: 'claude',
      format: 'claude_project_json',
      scope: 'project',
      officialPath: path.join(workspaceRoot, '.mcp.json'),
      configPath: path.join(workspaceRoot, '.mcp.json'),
      pathLocked: true,
      docsUrl: 'https://code.claude.com/docs/en/mcp',
      verifyArgs: ['mcp', 'list'],
    },
    {
      id: 'gemini-cli',
      name: 'Gemini CLI',
      cliCommand: 'gemini',
      format: 'gemini_project_json',
      scope: 'project',
      officialPath: path.join(workspaceRoot, '.gemini', 'settings.json'),
      configPath: path.join(workspaceRoot, '.gemini', 'settings.json'),
      pathLocked: true,
      docsUrl: 'https://google-gemini.github.io/gemini-cli/docs/tools/mcp-server.html',
      verifyArgs: ['mcp', 'list', '--scope', 'project'],
    },
  ];
}

module.exports = {
  buildOfficialTools,
};
